File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/keyboard/CharacterSenderStrategy.java
package net.kdt.pojavlaunch.customcontrols.keyboard;

/** 必要なブリッジを介して文字を送信するためのシンプルなインターフェース */
public interface CharacterSenderStrategy {
    /** 削除する文字があるときに呼び出されます。連続して複数回呼び出される可能性があります */
    void sendBackspace();

    /** エンターキーを送信するときに呼び出されます */
    void sendEnter();

    /** 送信する文字があるときに呼び出されます。連続して複数回呼び出される可能性があります */
    void sendChar(char character);

}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/keyboard/CharacterSenderStrategy.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.keyboard

/** 文字入力をMinecraftエンジンに送信する戦略インターフェース。 */
interface CharacterSenderStrategy {
    /** バックスペースキーを送信します。 */
    fun sendBackspace()
    /** エンターキーを送信します。 */
    fun sendEnter()
    /**
     * 1文字を送信します。
     * @param character 送信する文字
     */
    fun sendChar(character: Char)
}